platform/ext/target/nordic_nrf/common/native_drivers/spu.h

```diff
@@ -21,6 +21,8 @@
 #include <stdint.h>
 #include <stdbool.h>
 
+#include "Driver_Common.h"
```

Comment:
Well, I thought it is the best way to take the ARM driver enums defined, e.g. ARM_DRIVER_OK,
any alternatives?

Reply:
I guess it's about `ARM_DRIVER_OK` not defined in `target_cfg.c`. But the question is if you in fact need it there? It seems that `spu_init_cfg()` and `spu_periph_init_cfg()` could use other error codes, not the CMSIS driver ones.

Reply:
I would like to preserve uniformity, and consistency in `spm_hal.c` function `tfm_spm_hal_init_isolation_hw()` (which checks that the _init_cfg() functions return ARM_DRIVER_OK), across the various supported platforms.

If spu_init_cfg() and spu_periph_init_cfg() would return other error codes (e.g. Nordic specific), then the spm_hal would need to read these other error codes.

If this is preferable, let me know. I don't care so much about the error codes here as they are not propagated in higher levels.

As a side note, since the MPC (and PPC) _are actually_ CMSIS drivers, I would think that in the "near" future, our spu.c chould also be made a CMSIS driver and then it would make sense to return the ARM_DRIVER_OK

Reply:
> I would like to preserve uniformity, and consistency in `spm_hal.c` function `tfm_spm_hal_init_isolation_hw()` (which checks that the _init_cfg() functions return ARM_DRIVER_OK), across the various supported platforms.
> 
> If spu_init_cfg() and spu_periph_init_cfg() would return other error codes (e.g. Nordic specific), then the spm_hal would need to read these other error codes.

AN539 uses `enum tfm_plat_err_t` for those. To me it looks more adequate.
https://github.com/NordicPlayground/trusted-firmware-m/blob/94f4fb3510acaabf430781d45aa40267fa9ac34d/platform/ext/target/mps2/an539/spm_hal.c#L36-L53

Reply:
@anangl  AN539 is the only one that returns TFM_PLAT_ERR codes.
But then I think you might be right here. Right now nRF SPU is not a CMSIS "driver" (yet) so it might make more sense to use the TFM Platform Error codes directly.

As i mentioned above, my only intention was to preserve uniformity, but since there is already one platform deviating, it does not harm to have nRF ones doing the same.

Will update based on your proposal